test/Winton.Extensions.Configuration.Consul.Test/ConsulConfigurationProviderTests.cs

```diff
@@ -87,18 +87,23 @@ private async Task ShouldCancelPollingTaskWhenReloading()
                             }
 
                             expectedKvCalls++;
-                            if (cancellationToken == default)
+                            if (token.CanBeCanceled)
                             {
-                                cancellationToken = token;
+                                token.Register(() => pollingCancelled.TrySetResult(true));
                             }
                         })
-                    .Returns(
-                        pollingCancelled.Task.IsCompleted
-                            ? Task.FromCanceled<QueryResult<KVPair[]>>(pollingCancelled.Task.Result)
-                            : Task.FromResult(new QueryResult<KVPair[]> { StatusCode = HttpStatusCode.OK }));
+                    .Returns<string, QueryOptions, CancellationToken>(
+                        (_, __, token) =>
```

Comment:
You can do away with the `return` here and do:

```
(_, __, token) => token.IsCancellationRequested ?
    ? Task.FromCanceled<QueryResult<KVPair[]>>(token)
    : Task.Delay(5).ContinueWith(t => new QueryResult<KVPair[]> { StatusCode = HttpStatusCode.OK })
```

Reply:
You've got it! 👍